src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,85 @@
+package duke;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.FileWriter;
+import java.io.File;
+
+/**
+ * The Storage class deals with loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+    private String filePath;
+
+    /**
+     * Constructor for a Storage object.
+     * @param filePath filePath of the file
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads the data of the file given and returns it in an ArrayList.
+     * @return An ArrayList of tasks stored in the file.
+     * @throws DukeException if file is not found.
+     */
+    public ArrayList<Task> load() throws DukeException {
```

Comment:
This method is more than 30 lines and very lengthy, perhaps you can consider shortening it to improve readability?

Reply:
Alright thanks